File: src/models/assignment-return-kind.ts
export const AssignmentReturnKind = {
  None: 0,
  Paper: 1,
  FileUpload: 2,
  Kiosk: 3,

  /** Only available since version 2024. */
  AudioRecording: 4
} as const;

export type AssignmentReturnKind = typeof AssignmentReturnKind[keyof typeof AssignmentReturnKind];

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/AssignmentReturnKind.kt
package ink.literate.pawnote.models

enum class AssignmentReturnKind (val code: Int) {
    None(0),
    Paper(1),
    FileUpload(2),
    Kiosk(3),

    /** Only available since version 2024. */
    AudioRecording(4);

    companion object {
        fun fromInt (code: Int) = entries.first { it.code == code }
    }
}